heron/metricscachemgr/tests/java/com/twitter/heron/metricscachemgr/metricscache/MetricsCacheQueryUtilsTest.java

```diff
@@ -0,0 +1,281 @@
+// Copyright 2016 Twitter. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.twitter.heron.metricscachemgr.metricscache;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import com.twitter.heron.metricscachemgr.metricscache.datapoint.ExceptionDatapoint;
+import com.twitter.heron.proto.system.Common;
+import com.twitter.heron.proto.tmaster.TopologyMaster;
+
+import static com.twitter.heron.metricscachemgr.metricscache.MetricsCacheQueryUtils.toProtobuf;
+
+public class MetricsCacheQueryUtilsTest {
+  @Test
+  public void testFromProtoBufMetricInterval() {
+    TopologyMaster.MetricRequest request =
+        TopologyMaster.MetricRequest.newBuilder()
+            .setComponentName("c1")
+            .addInstanceId("i1").addInstanceId("i2")
+            .addMetric("m1").addMetric("m2")
+            .setInterval(100)// in seconds
+            .build();
+
+    MetricsCacheQueryUtils.MetricRequest request1 =
+        MetricsCacheQueryUtils.fromProtobuf(request);
+
+    Assert.assertEquals(1, request1.componentNameInstanceId.keySet().size());
+    Assert.assertEquals(true, request1.componentNameInstanceId.containsKey("c1"));
+    Assert.assertEquals(2, request1.componentNameInstanceId.get("c1").size());
+    Assert.assertEquals(true, request1.componentNameInstanceId.get("c1").contains("i1"));
+    Assert.assertEquals(true, request1.componentNameInstanceId.get("c1").contains("i2"));
+    Assert.assertEquals(100 * 1000, request1.endTime - request1.startTime); // in milli-seconds
+    Assert.assertEquals(MetricsCacheQueryUtils.Granularity.AGGREGATE_ALL_METRICS,
+        request1.aggregationGranularity);
+  }
+
+  @Test
+  public void testFromProtoBufMetricExplicitInterval() {
+    TopologyMaster.MetricRequest request =
+        TopologyMaster.MetricRequest.newBuilder()
+            .setComponentName("c1")
+            .addInstanceId("i1").addInstanceId("i2")
+            .addMetric("m1").addMetric("m2")
+            .setExplicitInterval(TopologyMaster.MetricInterval.newBuilder()
+                .setStart(100).setEnd(200)) // in seconds
+            .build();
+
+    MetricsCacheQueryUtils.MetricRequest request1 =
+        MetricsCacheQueryUtils.fromProtobuf(request);
+
+    Assert.assertEquals(1, request1.componentNameInstanceId.keySet().size());
```

Comment:
Same comment as the other test class above. Please refactor repeated assertion logic into helper methods to reduce verbosity.